haystack/utils/device.py

```diff
@@ -1,34 +1,471 @@
+
+    def to_torch_str(self) -> str:
+        """
+        Convert the component device representation to PyTorch string format.
+        Device maps are not supported.
+
+        :returns:
+            The PyTorch device string representation.
+        """
+        if self._single_device is None:
+            raise ValueError("Only single devices can be converted to PyTorch format")
+
+        assert self._single_device is not None
+        return str(self._single_device)
+
+    def to_spacy(self) -> int:
+        """
+        Convert the component device representation to spaCy format.
+        Device maps are not supported.
+
+        :returns:
+            The spaCy device representation.
+        """
+        if self._single_device is None:
+            raise ValueError("Only single devices can be converted to spaCy format")
+
+        assert self._single_device is not None
+        if self._single_device.type == DeviceType.GPU:
+            assert self._single_device.id is not None
+            return self._single_device.id
+        else:
+            return -1
+
+    # pylint: disable=inconsistent-return-statements
+    def to_hf(self) -> Union[Union[int, str], Dict[str, Union[int, str]]]:
+        """
+        Convert the component device representation to HuggingFace format.
+
+        :returns:
+            The HuggingFace device representation.
+        """
+
+        def convert_device(device: Device, *, gpu_id_only: bool = False) -> Union[int, str]:
+            if gpu_id_only and device.type == DeviceType.GPU:
+                assert device.id is not None
+                return device.id
+            else:
+                return str(device)
+
+        if self._single_device is not None:
+            return convert_device(self._single_device)
+        elif self._multiple_devices is not None:
+            return {
+                key: convert_device(device, gpu_id_only=True) for key, device in self._multiple_devices.mapping.items()
+            }
+        else:
+            # Unreachable
+            assert False
+
+    def update_hf_kwargs(self, hf_kwargs: Dict[str, Any], *, overwrite: bool) -> Dict[str, Any]:
+        """
+        Convert the component device representation to HuggingFace format
+        and add them as canonical keyword arguments to the keyword arguments
+        dictionary.
+
+        :param hf_kwargs:
+            The HuggingFace keyword arguments dictionary.
+        :param overwrite:
+            Whether to overwrite existing device arguments.
+        :returns:
+            The HuggingFace keyword arguments dictionary.
+        """
+        if not overwrite and any(x in hf_kwargs for x in ("device", "device_map")):
+            return hf_kwargs
+
+        converted = self.to_hf()
+        key = "device_map" if self.multiple_devices else "device"
+        hf_kwargs[key] = converted
+        return hf_kwargs
+
+    @property
+    def multiple_devices(self) -> bool:
```

Comment:
I find this name a little confusing for a boolean property. How about `has_multiple_devices` or something on this lines, that implies a yes/no answer? Not a blocker though.